### PR TITLE
PS-281: Convert LP 1354988 fix to new "SHOW EFFECTIVE GRANTS"

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: sql/sql_yacc.yy [hunk 1]
```diff
@@ -1244,6 +1244,7 @@ void warn_about_deprecated_national(THD *thd)
 %token<keyword> TOKU_SMALL_SYM
 %token<keyword> TOKU_DEFAULT_SYM
 %token<keyword> USER_STATS_SYM
+%token<keyword> EFFECTIVE_SYM
```
Comment (laurynas-biveinis) [inline sql/sql_yacc.yy]: Add a header comment "Tokens from Percona Server 8.0" for this one

File: sql/auth/sql_authorization.cc [hunk 2]
```diff
@@ -1184,8 +1185,12 @@ void get_sp_access_map(
     */
 
     if (!strcmp(acl_user->user, user) &&
-        grant_proc->host.compare_hostname(acl_user->host.get_host(),
-                                          acl_user->host.get_host())) {
+        ((!effective_grants &&
```
Comment (laurynas-biveinis) [inline sql/auth/sql_authorization.cc]: The condition (!effective_grants && !my_strcasecmp()) || (effective_grants && compare_hostname()) is repeated several times - please extract (and remove one of the "effective_grants" in effective_grants || !effective_grants)

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: sql/auth/sql_authorization.cc [hunk 3]
```diff
@@ -1164,10 +1164,19 @@ void make_table_privilege_statement(THD *thd, ACL_USER *role,
   }
 }
 
+static bool hosts_match_for_grants(ACL_HOST_AND_IP &acl_db_host,
```
Comment (laurynas-biveinis) [inline sql/auth/sql_authorization.cc:1167]: Can this arg be const ref?

**Review by laurynas-biveinis — APPROVED** [review]

File: sql/auth/sql_authorization.cc (hunk 3 above)
Comment (inikep) [reply sql/auth/sql_authorization.cc:1167]: no

**Pull request merged**